buttercup.el

```diff
@@ -1536,46 +1548,55 @@ EVENT and ARG are described in `buttercup-reporter'."
        (let ((defined (buttercup-suites-total-specs-defined arg))
              (pending (buttercup-suites-total-specs-pending arg)))
          (if (> pending 0)
              (buttercup--print "Running %s out of %s specs.\n\n"
                                (- defined pending)
                                defined)
-           (buttercup--print "Running %s specs.\n\n" defined))))
+           (buttercup--print "Running %s specs.\n\n" defined)))
+       (buttercup--flush-pending-output))
 
       (`suite-started
        (let ((level (length (buttercup-suite-or-spec-parents arg))))
+         (buttercup--start-output-level)
          (buttercup--print "%s%s\n"
                            (make-string (* 2 level) ?\s)
                            (buttercup-suite-description arg))))
 
       (`spec-started
        (let ((level (length (buttercup-suite-or-spec-parents arg))))
+         (buttercup--start-output-level)
          (buttercup--print "%s%s"
                            (make-string (* 2 level) ?\s)
                            (buttercup-spec-description arg))))
 
       (`spec-done
-       (cond
-        ((eq (buttercup-spec-status arg) 'passed)) ; do nothing
-        ((eq (buttercup-spec-status arg) 'failed)
-         (buttercup--print "  FAILED")
-         (setq buttercup-reporter-batch--failures
-               (append buttercup-reporter-batch--failures
-                       (list arg))))
-        ((eq (buttercup-spec-status arg) 'pending)
-         (buttercup--print "  %s" (buttercup-spec-failure-description arg)))
-        (t
-         (error "Unknown spec status %s" (buttercup-spec-status arg))))
-       (buttercup--print " (%s)\n"
-                         (seconds-to-string
-                          (float-time (buttercup-elapsed-time arg)))))
+       (if (buttercup-spec-omitted-p arg)
+           (buttercup--discard-one-output-level)
+         (cond
+          ((eq (buttercup-spec-status arg) 'passed)) ; do nothing
+          ((eq (buttercup-spec-status arg) 'failed)
+           (buttercup--print "  FAILED")
+           (setq buttercup-reporter-batch--failures
+                 (append buttercup-reporter-batch--failures
+                         (list arg))))
+          ((eq (buttercup-spec-status arg) 'pending)
+           (buttercup--print "  %s" (buttercup-spec-failure-description arg)))
+          (t
+           (error "Unknown spec status %s" (buttercup-spec-status arg))))
+         (buttercup--print " (%s)\n"
+                           (seconds-to-string
+                            (float-time (buttercup-elapsed-time arg))))
+         (buttercup--flush-pending-output)))
 
       (`suite-done
-       (when (= 0 (length (buttercup-suite-or-spec-parents arg)))
-         (buttercup--print "\n")))
+       (unless (buttercup--discard-one-output-level)
+         (when (= 0 (length (buttercup-suite-or-spec-parents arg)))
+           (buttercup--print "\n"))))
```

Comment:
(or (buttercup--discard-one-output-level)
     (< 0 (length ....
     (buttercup--print  ...
The `=` sexp can be replaced with `(null (buttercup-suite-or-spec-parent arg))`.

Reply:
It's a matter of style, I thought it would be more explicit this way.